### ISSUE TITLE
[native-image] Undefined reference to `Java_sun_misc_Unsafe_defineClass'

### ISSUE THREAD

**Issue by lburgazzoli** [oracle/graal#1975]
Related to https://github.com/oracle/graal/issues/1971, we are also experiencing this issue:

```
INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] /opt/data/sfw/lang/java/8-graal/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:NativeLinkerOption=-no-pie -H:+JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]    classlist:   7,951.00 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]        (cap):     740.18 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]        setup:   2,188.86 ms
15:25:17,276 INFO  [org.ecl.jet.uti.log] Logging initialized @10585ms to org.eclipse.jetty.util.log.Slf4jLog
15:25:19,042 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Final
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]   (typeflow):  40,757.55 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]    (objects):  21,194.85 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]   (features):   1,273.53 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]     analysis:  67,374.22 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]     (clinit):   1,439.31 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]     universe:   3,921.49 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]      (parse):   4,231.95 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]     (inline):   6,405.22 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]    (compile):  34,405.39 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]      compile:  48,118.82 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]        image:   5,186.32 ms
[camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner:77025]        write:     629.01 ms
Fatal error: java.lang.RuntimeException: java.lang.RuntimeException: host C compiler or linker does not seem to work: java.lang.RuntimeException: returned 1

Running command: cc -v -o /home/lburgazz/work/apache/dev/camel-quarkus/integration-tests/salesforce/target/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-native-image-source-jar/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner -z noexecstack -Wl,--gc-sections -Wl,--dynamic-list -Wl,/tmp/SVM-3705449318764966532/exported_symbols.list -Wl,-x -L/tmp/SVM-3705449318764966532 -L/opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib -L/opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64 /tmp/SVM-3705449318764966532/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner.o /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/libffi.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/liblibchelper.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/libjvm.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/libstrictmath.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libnet.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libjava.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libzip.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libnio.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libsunec.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libsunec.a -lm -lpthread -ldl -lpthread -lz -lrt -lstdc++ -no-pie

Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,d,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 9.2.1 20190827 (Red Hat 9.2.1-1) (GCC) 
COMPILER_PATH=/usr/libexec/gcc/x86_64-redhat-linux/9/:/usr/libexec/gcc/x86_64-redhat-linux/9/:/usr/libexec/gcc/x86_64-redhat-linux/:/usr/lib/gcc/x86_64-redhat-linux/9/:/usr/lib/gcc/x86_64-redhat-linux/
LIBRARY_PATH=/usr/lib/gcc/x86_64-redhat-linux/9/:/usr/lib/gcc/x86_64-redhat-linux/9/../../../../lib64/:/lib/../lib64/:/usr/lib/../lib64/:/usr/lib/gcc/x86_64-redhat-linux/9/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-v' '-o' '/home/lburgazz/work/apache/dev/camel-quarkus/integration-tests/salesforce/target/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-native-image-source-jar/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner' '-z' 'noexecstack' '-L/tmp/SVM-3705449318764966532' '-L/opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib' '-L/opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64' '-no-pie' '-mtune=generic' '-march=x86-64'
 /usr/libexec/gcc/x86_64-redhat-linux/9/collect2 -plugin /usr/libexec/gcc/x86_64-redhat-linux/9/liblto_plugin.so -plugin-opt=/usr/libexec/gcc/x86_64-redhat-linux/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccZQxE1V.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --no-add-needed --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o /home/lburgazz/work/apache/dev/camel-quarkus/integration-tests/salesforce/target/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-native-image-source-jar/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner -z noexecstack /usr/lib/gcc/x86_64-redhat-linux/9/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/9/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/9/crtbegin.o -L/tmp/SVM-3705449318764966532 -L/opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib -L/opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64 -L/usr/lib/gcc/x86_64-redhat-linux/9 -L/usr/lib/gcc/x86_64-redhat-linux/9/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/9/../../.. --gc-sections --dynamic-list /tmp/SVM-3705449318764966532/exported_symbols.list -x /tmp/SVM-3705449318764966532/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner.o /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/libffi.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/liblibchelper.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/libjvm.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/svm/clibraries/linux-amd64/libstrictmath.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libnet.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libjava.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libzip.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libnio.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libsunec.a /opt/data/sfw/lang/java/graalvm-ce-java8-19.3.0/jre/lib/libsunec.a -lm -lpthread -ldl -lpthread -lz -lrt -lstdc++ -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-redhat-linux/9/crtend.o /usr/lib/gcc/x86_64-redhat-linux/9/../../../../lib64/crtn.o
/usr/bin/ld: /tmp/SVM-3705449318764966532/camel-quarkus-integration-test-salesforce-1.0.0-SNAPSHOT-runner.o:(.data+0x1050): undefined reference to `Java_sun_misc_Unsafe_defineClass'
collect2: error: ld returned 1 exit status

```

to reproduce:

    GraalVM 19.3.0 Java 8
    get this PR: apache/camel-quarkus#537
    mvn clean install -DskipTests -DskipITs
    mvn clean install -Dnative -rf :camel-quarkus-integration-test-salesforce

**Comment by olpaw** [comment]
This looks like a duplicate of https://github.com/oracle/graal/issues/1875
Please check if you can reproduce the issue on master.

**Comment by lburgazzoli** [comment]
is there a build I can download to test ?

**Comment by olpaw** [comment]
We don't have nightly builds yet. Please build one yourself from master
```
cd graal/vm
mx --env ce build
```

**Comment by plokhotnyuk** [comment]
Wow! It is `20.0.0-dev`, not `19.3.x`!

**Comment by lburgazzoli** [comment]
I didn0t rebuild quarkus against master but repplacing the native.image binary with the one from master fixes the issue

**Comment by olpaw** [comment]
The fix for #1875 will also get backported to 19.3.1
